File: scripts/perl/input.pl
#!usr/local/bin/perl

# Print Menu
print_menu();

# Get user input
get_usr_input();

# Subroutines
sub get_usr_input {
	printf("input> ");
	$result = <STDIN>;
	chomp $result;
	if ( $result eq "1" ) {
		print_menu();
		get_usr_input();
	}
	elsif ( $result eq "2" ) {
		printf("Help is coming soon!\n");
		get_usr_input();
	}
	elsif ( $result eq "3" ) {
		printf( "Current user: root\n");
		get_usr_input();
	}
	elsif ( $result eq "4" ) {
		printf("1) add(x,y) -> r=x+y\n2) sub(x,y) -> r=x-y");
		printf("\ninput/examples> ");
		$answer = <STDIN>;
		if ( $answer eq "1" ) {
			my $x = int( rand(1000) ), $y = int( rand(1000) );
			my $r = $x + $y ;
			printf( "x: ". $x . "\ny: " . $y );
			printf( "\nr: ". $r);
			printf("\n");
		}
		elsif ( $answer eq "2" ) {
			my $x = int( rand(1000) ), $y = int( rand(1000) );
			my $r = $x - $y ;
			printf( "x: ". $x . "\ny: " . $y );
			printf( "\nr: ". $r);
			printf("\n");
		}
		else {
			printf("Wrong input.\n");
		}
		get_usr_input();
	}
	elsif ( $result eq "5" ) {
		printf("Terminating...\nAre you sure you want to quit? y/n\ninput/exit> ");
		$an = <STDIN>;
		chomp($an);
		if(($an eq "y") || ($an eq "Y") ){
			printf("Exiting...\n");
			exit;
		}else{
			printf("Aborted...\n");
		}
		get_usr_input();
	}
	else {
		printf("Wrong input.\n");
		get_usr_input();
	}
}

sub print_menu {
	printf("\n");
	printf("##########################################\n");
	printf("############ PERL INPUT TEST #############\n");
	printf("##########################################\n");
	printf("#### COMMANDS                         ####\n");
	printf("####                                  ####\n");
	printf("#### 1)Main menu                      ####\n");
	printf("#### 2)Help                           ####\n");
	printf("#### 3)User                           ####\n");
	printf("#### 4)PERL Examples                  ####\n");
	printf("#### 5)Exit                           ####\n");
	printf("##########################################\n");
}

File: scripts/python/input.py

print; 
print;
print "##########################################"
print "############ PYTHON INPUT TEST ###########"
print "##########################################"
print "#### COMMANDS                         ####"
print "####                                  ####"
print "#### 1)Help                           ####"
print "#### 2)User                           ####"
print "#### 3)Python Examples                ####"
print "#### 4)Exit                           ####"
print "##########################################"
usr_inpt = raw_input("input> ")

# Add function
def add():
 from random import randint
 x = randint(0,100)
 y = randint(0,100)
 print "x:",x,"\ny:", y
 return "Result",x+y

# Sub function
def sub():
 from random import randint
 x = randint(0,100)
 y = randint(0,100)
 print "x:",x,"\ny:", y
 return "Result",x-y



def print_input(usr_inpt):
 var=usr_inpt
 if var == "1":
   print "Help is coming soon!"
   print_input(raw_input("input> "))
 elif var == "2":
   print "root"
   print_input(raw_input("input> "))
 elif var == "3":
   print "1) add(x,y) -> r=x+y\n2) sub(x,y) -> r=x-y"
   ex = raw_input("input/example> ")
   if ex == "1":
    print add()
    print_input(raw_input("input> "))
   elif ex == "2":
    print sub()
    print_input(raw_input("input> "))
   else:
    print_input(raw_input("Wrong input.\ninput> "))
 elif var == "4": 
   print "Terminating..."  
   yn = raw_input("Are you sure you want to quit? y/n> ")   
   if yn == "y": 
    print "Success."     
    import sys
    sys.exit()
   elif yn == "n":
    print "Aborted."
    print_input(raw_input("input> "))
   else:
    print "Wrong input."
    print_input(raw_input("input> "))
 else:
   print "Wrong input."
   print_input(raw_input("input> "))
 return
 
print_input(usr_inpt)

